### PR TITLE
Fix the check for existing users when an email is entered in the ium …

### PR DESCRIPTION
**Pull request by b12f**
…principal search

### CODE REVIEW

**Review by wielinde — APPROVED** [review]
I tested the behavior. Seems to be better now.

**Comment by wielinde** [comment]
@b12f One little thing: If we follow our naming convention for observables right then the method `loadPrincipalData` should be called `loadPrincipalData$`, am I right?

File: frontend/src/app/modules/invite-user-modal/principal/principal-search.component.ts [hunk 1]
```diff
@@ -49,12 +50,12 @@ export class PrincipalSearchComponent extends UntilDestroyedMixin implements OnI
     this.input$,
     this.currentUserService.hasCapabilities$('users/create'),
   ).pipe(
-    map(([elements, input, canCreateUsers]) =>
-      canCreateUsers
+    map(([elements, input, canCreateUsers]) => {
+      return canCreateUsers
```
Comment (wielinde) [inline frontend/src/app/modules/invite-user-modal/principal/principal-search.component.ts:54]: Ensure returned type is `boolean`
```suggestion
      return !!canCreateUsers
```

File: frontend/src/app/modules/invite-user-modal/principal/principal-search.component.ts [hunk 2]
```diff
@@ -49,12 +50,12 @@ export class PrincipalSearchComponent extends UntilDestroyedMixin implements OnI
...
       && this.type === PrincipalType.User
       && input?.includes('@')
```
Comment (wielinde) [inline frontend/src/app/modules/invite-user-modal/principal/principal-search.component.ts:56]: Ensure returned type is `boolean`
```suggestion
      && !!input?.includes('@')
```

**Comment by oliverguenther** [comment]
> @b12f One little thing: If we follow our naming convention for observables right then the method `loadPrincipalData` should be called `loadPrincipalData$`, am I right?

We only set the `$` sign to variables to clarify their use in templates. It shouldn't set to methods

**Pull request merged**